### PR TITLE
Add email verification mechanism using nodemailer

### LINKED ISSUE #8: Add email verification during creating a new account.
If a new user create an account for the first time, a link should be sent in the email. And only after clicking on the link and verifying their email, user can move forward.

### PR DESCRIPTION
**Pull request by IrfanSalim**
Closes #8 

Used nodeMailer to send verification emails, mandated users to verify accounts before logging in [screenshot below]

image

Sample email:
image

Note: Please add the email and password to be used for sending the email to the env file and make sure that the account is allowed to login without verification, if 2FA is enabled for Gmail please create an app password and add that in the env file. For more detailed check - https://nodemailer.com/usage/using-gmail/

### CODE REVIEW

File: BackEnd/server.js [hunk 1]
```diff
@@ -23,6 +25,77 @@ async function idExists(db, tableName, ID_key, ID) {
...
+                }
+              </style>
+            </head>
+            <body>
+              <div class="container">
+                <h1>Hello ${name},</h1>
+                <p>Thank you for signing up with AnimeHub! To get started, please verify your email by clicking the button below:</p>
+                <a href="http://localhost:5000/verify/${token}">Verify Your Email</a>
+                <p>If the button above doesn't work, you can also copy and paste the following link into your web browser:</p>
+                <p><a href="http://localhost:5000/verify/${token}">http://localhost:5000/verify/${token}</a></p>
```
Comment (moonpatel) [inline BackEnd/server.js]: @IrfanSalim do not harcode hostname (localhost:5000) instead use environment variables. Remove the hardcoded values from all the places.

Comment (IrfanSalim) [reply BackEnd/server.js]: Done @moonpatel

**Comment by IrfanSalim** [comment]
Hey, @Peacexoom can you check and merge this, please?

I will start working on the hashing issue once this is closed.

**Comment by moonpatel** [comment]
> Hey, @Peacexoom can you check and merge this, please?
> 
> I will start working on the hashing issue once this is closed.

Hey @IrfanSalim everything is working fine, just one small change add the FRONTEND, BACKEND, EMAIL and EMAIL_PASSWORD fields in .env.example with documentation.
Then I will merge your PR.

**Comment by IrfanSalim** [comment]
Done @moonpatel

**Comment by moonpatel** [comment]
> Done @moonpatel

Thanks for your contribution @IrfanSalim !

**Pull request merged**